pkg/actuators/machine/actuator_test.go

```diff
@@ -684,41 +639,58 @@ func TestAvailabiltyZone(t *testing.T) {
 				t.Fatalf("Could not create AWS machine actuator: %v", err)
 			}
 
-			mockRunInstancesForPlacement(mockAWSClient, tc.availabilityZone, tc.subnet)
-			mockDescribeInstances(mockAWSClient, false)
-			mockTerminateInstances(mockAWSClient)
-			mockRegisterInstancesWithLoadBalancer(mockAWSClient, false)
-			mockDescribeSubnets(mockAWSClient)
+			var placement *ec2.Placement
+			if tc.availabilityZone != "" && tc.subnet == "" {
+				placement = &ec2.Placement{AvailabilityZone: aws.String(tc.availabilityZone)}
+			}
 
-			actuator.Create(context.TODO(), cluster, machine)
-		})
-	}
-}
+			mockAWSClient.EXPECT().RunInstances(Placement(placement)).Return(
+				&ec2.Reservation{
+					Instances: []*ec2.Instance{
+						{
+							ImageId:    aws.String("ami-a9acbbd6"),
+							InstanceId: aws.String("i-02fcb933c5da7085c"),
+							State: &ec2.InstanceState{
+								Name: aws.String("Running"),
+								Code: aws.Int64(16),
+							},
+							LaunchTime: aws.Time(time.Now()),
+						},
+					},
+				}, nil)
 
-func mockDescribeSubnets(mockAWSClient *mockaws.MockClient) {
-	mockAWSClient.EXPECT().DescribeSubnets(gomock.Any()).Return(&ec2.DescribeSubnetsOutput{}, nil)
-}
+			mockAWSClient.EXPECT().DescribeInstances(gomock.Any()).Return(
+				&ec2.DescribeInstancesOutput{
+					Reservations: []*ec2.Reservation{
+						{
+							Instances: []*ec2.Instance{
+								{
+									ImageId:    aws.String("ami-a9acbbd6"),
+									InstanceId: aws.String("i-02fcb933c5da7085c"),
+									State: &ec2.InstanceState{
+										Name: aws.String("Running"),
+										Code: aws.Int64(16),
+									},
+									LaunchTime: aws.Time(time.Now()),
+								},
+							},
+						},
+					},
+				}, nil).AnyTimes()
 
-func mockRunInstancesForPlacement(mockAWSClient *mockaws.MockClient, availabilityZone, subnet string) {
-	var placement *ec2.Placement
-	if availabilityZone != "" && subnet == "" {
-		placement = &ec2.Placement{AvailabilityZone: aws.String(availabilityZone)}
-	}
+			mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(&ec2.TerminateInstancesOutput{}, nil)
 
-	mockAWSClient.EXPECT().RunInstances(Placement(placement)).Return(
-		&ec2.Reservation{
-			Instances: []*ec2.Instance{
-				{
-					ImageId:    aws.String("ami-a9acbbd6"),
-					InstanceId: aws.String("i-02fcb933c5da7085c"),
-					State: &ec2.InstanceState{
-						Name: aws.String("Running"),
-						Code: aws.Int64(16),
-					},
-					LaunchTime: aws.Time(time.Now()),
-				},
-			},
-		}, nil)
+			// RegisterInstancesWithLoadBalancer should be called for every load balancer name in the machine
+			// spec for create and for update (3 * 2 = 6)
+			for i := 0; i < 6; i++ {
```

Comment:
Can this upper bound be statically inferred?

Reply:
I don't think it matters here how many times the method is called. Will remove the loop.